hexrd/WPPF.py

```diff
@@ -463,46 +494,80 @@ def _readHDF(self, fhdf, xtal):
         # else:
         #   raise IOError('material file does not exist.')
 
-        gid         = fid.get(xtal)
+        gid = fid.get(xtal)
 
-        self.sgnum       = np.asscalar(np.array(gid.get('SpaceGroupNumber'), \
-                                     dtype = np.int32))
-        self.sgsetting = np.asscalar(np.array(gid.get('SpaceGroupSetting'), \
-                                        dtype = np.int32))
-        self.sgsetting -= 1
+        self.sgnum = np.asscalar(np.array(gid.get('SpaceGroupNumber'),
+                                          dtype=np.int32))
+        self.sgsetting = np.asscalar(np.array(gid.get('SpaceGroupSetting'),
+                                              dtype=np.int32))
         """
             IMPORTANT NOTE:
-            note that the latice parameters in EMsoft is nm by default
+            note that the latice parameters is nm by default
             hexrd on the other hand uses A as the default units, so we
             need to be careful and convert it right here, so there is no
             confusion later on
         """
-        self.lparms      = list(gid.get('LatticeParameters'))
+        self.lparms = list(gid.get('LatticeParameters'))
         fid.close()
 
+    def _init_from_materials(self, material_obj):
+        '''
+        this function is used to initialize the materials_lebail class
+        from an instance of the material.Material class. this option is
+        provided for easy integration of the hexrdgui with WPPF.
+        '''
+        self.dmin = material_obj.dmin.value
```

Comment:
Do we need to make sure this is in nanometers? We are using angstroms in the GUI, and setting `_angstroms()` on the material's `dmin`.

Reply:
yep, nice catch.